projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts

```diff
@@ -191,6 +191,25 @@ describe('IgxGrid - Grid Sorting #grid', () => {
             expect(grid.getCellByColumn(0, currentColumn).value.toISOString()).toEqual('2017-06-19T11:43:07.714Z');
         });
 
+        it('Should not mutate original data when sorting date column', () => {
+            fixture = TestBed.createComponent(GridWithPrimaryKeyComponent);
+            fixture.componentInstance.data = SampleTestData.personJobDataFull().map(rec => {
+                return Object.assign({}, { ...rec, HireDate: new Date(rec.HireDate) });
+            });
+            fixture.detectChanges();
+
+            grid = fixture.componentInstance.grid;
+            const hireDateCol = grid.columns.findIndex(col => col.field === "HireDate");
+            grid.columns[hireDateCol].dataType = 'date';
+            fixture.detectChanges();
+
+            grid.sort({ fieldName: 'HireDate', dir: SortingDirection.Asc });
+            fixture.detectChanges();
+
+            const timeParts = (date: Date) => date.getHours() + date.getMinutes() + date.getSeconds() + date.getMinutes();
+            expect(grid.data.every(rec => timeParts(rec.HireDate) === 0)).toEqual(false);
```

Comment:
This test is not testing if the value has been mutated or not, it tests if the sum of hours, minutes, etc, is 0. It will produce false positive, if the value coming from the dataset is "Thu Jul 06 2023 00:00:00 GMT+0300 (Eastern European Summer Time)" for example.

Reply:
P.S. Ok the expectation is verified against `every` , I get it.